plugins/fastem_script_runner.py

```diff
@@ -0,0 +1,635 @@
+        f"not found within '{base_path}' (and its subdirectories) for version '{version}'."
+    )
+
+
+class WxTextCtrlOutput:
+    """Custom output stream to redirect text output to a wx.TextCtrl."""
+
+    def __init__(self, text_ctrl: wx.TextCtrl, style: Optional[wx.TextAttr] = None):
+        """
+        :param text_ctrl: The wx.TextCtrl to which output will be directed.
+        :param style: Optional wx.TextAttr style for the text.
+        """
+        self.text_ctrl = text_ctrl
+        self.style = style
+        self.default_style = text_ctrl.GetDefaultStyle()
+
+    def write(self, text: str):
+        """
+        Write text to the wx.TextCtrl, ensuring it is done on the main thread.
+        :param text: The text to write to the TextCtrl.
+        """
+        if not wx.IsMainThread():
+            wx.CallAfter(self._do_write, text)
+            return
+        self._do_write(text)
+
+    def _do_write(self, text: str):
+        """
+        Internal method to write text to the TextCtrl.
+        :param text: The text to write to the TextCtrl.
+        """
+        if not self.text_ctrl:
+            return
+        if self.style:
+            self._append_text_styled(text, self.style)
+        else:
+            self.text_ctrl.AppendText(text)
+
+        self.text_ctrl.ShowPosition(
+            self.text_ctrl.GetLastPosition()
+        )  # Auto-scroll to end
+
+    def _append_text_styled(self, text: str, style: wx.TextAttr):
+        """
+        Append text to the TextCtrl with a specific style.
+        :param text: The text to append.
+        :param style: The wx.TextAttr style to apply to the text.
+        """
+        if not self.text_ctrl:
+            return
+        start_pos = self.text_ctrl.GetLastPosition()
+        self.text_ctrl.AppendText(text)
+        end_pos = self.text_ctrl.GetLastPosition()
+        self.text_ctrl.SetStyle(start_pos, end_pos, style)
+
+    def flush(self):
+        """
+        Flush the output stream. This is a no-op for wx.TextCtrl as it updates immediately.
+        """
+        pass  # wx.TextCtrl updates immediately
+
+
+class WxLogHandler(logging.Handler):
+    """Custom logging handler to redirect log messages to a wx.TextCtrl."""
+
+    def __init__(self, wx_text_ctrl_output_stream: WxTextCtrlOutput):
+        super().__init__()
+        self.wx_stream = wx_text_ctrl_output_stream
+        # You can set a default formatter here if you like
+        # Or allow it to be set externally
+        self.setFormatter(
+            logging.Formatter("%(asctime)s (%(module)s) %(levelname)s: %(message)s")
+        )
+
+    def emit(self, record):
+        # The `record` object has all the log information
+        # `self.format(record)` will turn it into a string based on the handler's formatter
+        try:
+            msg = self.format(record) + "\n"  # Add a newline like a typical console log
+            self.wx_stream.write(msg)
+        except Exception:
+            self.handleError(
```

Comment:
except Exception:
           # Default error handling (prints to original stderr)
            self.handleError(record)